### PR TITLE
#2021 - Refactor curation code

### PR DESCRIPTION
**Pull request by reckart**
**What's in the PR**
* Clean up code

**How to test manually**
* Use curation / no specific instructions

**Automatic testing**
* [ ] PR includes unit tests

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

File: inception-api/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/DocumentService.java [hunk 1]
```diff
@@ -489,8 +489,8 @@ CAS createOrReadInitialCas(SourceDocument aDocument, CasUpgradeMode aUpgradeMode
      * {@link SourceDocument}.
      * <p>
      * Note that this method does may not return an {@link AnnotationDocument annotation document}
-     * for every user in the project because they are created lazily when a user opens a document
-     * for annotation the first time.
+     * for every user in @Override the project because they are created lazily when a user opens a
```
Comment (jcklie) [inline inception-api/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/DocumentService.java:492]: Why is here a @Override ?

File: inception-ui-curation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/curation/component/model/AnnotatorSegment.java [hunk 2]
```diff
@@ -79,28 +76,7 @@ public void setAnnotatorState(AnnotatorState aState)
...
-    }
-
-    public void setSelectionByUsernameAndAddress(
-            Map<String, Map<Integer, AnnotationSelection>> aSelectionByUsernameAndAddress)
-    {
-        selectionByUsernameAndAddress = aSelectionByUsernameAndAddress;
-    }
-
-    public boolean equals(UserAnnotationSegment segment)
+    public boolean equals(AnnotatorSegment segment)
```
Comment (jcklie) [inline inception-ui-curation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/curation/component/model/AnnotatorSegment.java:79]: @Override ?

File: inception-ui-curation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/curation/component/model/BratSuggestionVisualizer.java [hunk 3]
```diff
@@ -62,7 +62,7 @@
     private @SpringBean ProjectService projectService;
     private @SpringBean UserDao userService;
 
-    public BratSuggestionVisualizer(String id, IModel<UserAnnotationSegment> aModel, int aPosition)
+    public BratSuggestionVisualizer(String id, IModel<AnnotatorSegment> aModel, int aPosition)
```
Comment (jcklie) [inline inception-ui-curation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/curation/component/model/BratSuggestionVisualizer.java:65]: aId?

Comment (reckart) [reply inception-ui-curation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/curation/component/model/BratSuggestionVisualizer.java]: Done.

File: inception-api/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/DocumentService.java (hunk 1 above)
Comment (reckart) [reply inception-api/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/DocumentService.java:492]: No clue, I'll remove it.

File: inception-ui-curation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/curation/component/model/AnnotatorSegment.java (hunk 2 above)
Comment (reckart) [reply inception-ui-curation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/curation/component/model/AnnotatorSegment.java:79]: I'm not sure if we even keep this class... also no idea what the "equals" does here - it does not override `equals(object)` - and if it did, we'd also have to override `hashCode`.

**Comment by UWinch** [comment]
Tested curation on curation page and sidebar, seems to work. However, `SearchServiceImpl - Error indexing annotation document` errors are being logged while annotating. Seems to also happen on master.

<details>

```

2021-02-26 12:29:47 ERROR [SYSTEM] SearchServiceImpl - Error indexing annotation document [barack_obama.txt](813) in project [MyFirstProject](21)
org.apache.lucene.index.IndexFormatTooNewException: Format version is not supported (resource BufferedChecksumIndexInput(MMapIndexInput(path="/srv/inception/repository/project/21/indexMtas/segments_3"))): 9 (needs to be between 6 and 8)
	at org.apache.lucene.codecs.CodecUtil.checkHeaderNoMagic(CodecUtil.java:216) ~[lucene-core-7.3.1.jar:7.3.1 ae0705edb59eaa567fe13ed3a222fdadc7153680 - caomanhdat - 2018-05-09 09:27:24]
	at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:306) ~[lucene-core-7.3.1.jar:7.3.1 ae0705edb59eaa567fe13ed3a222fdadc7153680 - caomanhdat - 2018-05-09 09:27:24]
	at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:290) ~[lucene-core-7.3.1.jar:7.3.1 ae0705edb59eaa567fe13ed3a222fdadc7153680 - caomanhdat - 2018-05-09 09:27:24]
	at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:1077) ~[lucene-core-7.3.1.jar:7.3.1 ae0705edb59eaa567fe13ed3a222fdadc7153680 - caomanhdat - 2018-05-09 09:27:24]
	at de.tudarmstadt.ukp.inception.search.index.mtas.MtasDocumentIndex.getIndexWriter(MtasDocumentIndex.java:224) ~[classes/:?]
	at de.tudarmstadt.ukp.inception.search.index.mtas.MtasDocumentIndex.getSearcherManager(MtasDocumentIndex.java:328) ~[classes/:?]
	at de.tudarmstadt.ukp.inception.search.index.mtas.MtasDocumentIndex.getTimestamp(MtasDocumentIndex.java:1067) ~[classes/:?]
	at de.tudarmstadt.ukp.inception.search.index.mtas.MtasDocumentIndex.indexDocument(MtasDocumentIndex.java:1133) ~[classes/:?]
	at de.tudarmstadt.ukp.inception.search.SearchServiceImpl.indexDocument(SearchServiceImpl.java:473) [classes/:?]
	at de.tudarmstadt.ukp.inception.search.SearchServiceImpl.indexDocument(SearchServiceImpl.java:426) [classes/:?]
	at de.tudarmstadt.ukp.inception.search.SearchServiceImpl$$FastClassBySpringCGLIB$$d6146f50.invoke(<generated>) [classes/:?]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) [spring-core-5.3.3.jar:5.3.3]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:779) [spring-aop-5.3.3.jar:5.3.3]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) [spring-aop-5.3.3.jar:5.3.3]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) [spring-aop-5.3.3.jar:5.3.3]
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123) [spring-tx-5.3.3.jar:5.3.3]
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388) [spring-tx-5.3.3.jar:5.3.3]
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119) [spring-tx-5.3.3.jar:5.3.3]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) [spring-aop-5.3.3.jar:5.3.3]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:750) [spring-aop-5.3.3.jar:5.3.3]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:692) [spring-aop-5.3.3.jar:5.3.3]
	at de.tudarmstadt.ukp.inception.search.SearchServiceImpl$$EnhancerBySpringCGLIB$$a0090e93.indexDocument(<generated>) [classes/:?]
	at de.tudarmstadt.ukp.inception.search.scheduling.tasks.IndexAnnotationDocumentTask.run(IndexAnnotationDocumentTask.java:47) [classes/:?]
	at de.tudarmstadt.ukp.inception.search.scheduling.TaskConsumer.run(TaskConsumer.java:70) [classes/:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]

```

</details>

**Comment by reckart** [comment]
The search errors are because you tested the dependency upgrade in between - this also upgrades the indexes. Either need to re-index or work with the updated dependencies.

**Comment by UWinch** [comment]
Ok, understood. Thanks for the explanation.

**Pull request merged**